### DISCUSSION TITLE
My getter implementation is being changed on load

### DISCUSSION THREAD

**Discussion by xenoterracide** [mikro-orm/mikro-orm#3900] [Q&A] [answered]
**Describe the bug**
Mongo implementation changing my getter

**To Reproduce**

```ts
import { Entity, PrimaryKey } from '@mikro-orm/core';
import * as uuid from 'uuid-mongodb';

@Entity({ abstract: true })
export abstract class BaseMongoAggregate<
    Id extends string,
  >
{
  constructor(id: uuid.MUUID = uuid.v4()) {
    super();
    this._id = id;
  }

  @PrimaryKey({ type: 'uuid' })
  protected _id: uuid.MUUID;

  get id(): Id {
    return this._id.toString() as Id;
  }

  protected set id(arg: Id) {
    this._id = uuid.from(arg);
  }
```

if you debug an instance returned by find an `MUUID` is returned, not a `string`, you can see that the getter implementation is not what I defined in the debugger.

<img width="741" alt="Screenshot 2023-01-04 at 11 01 11 AM" src="https://user-images.githubusercontent.com/5517/210597038-e7241c68-6f5a-4727-b858-6c430a2f0fc9.png">

**Expected behavior**
when calling `instance.id` it should use my getter, and not have the implementation replaced.

**Versions**

```
└─ services-core@workspace:.
   └─ @mikro-orm/mongodb@npm:5.6.3 [8316d] (via npm:^5.6.3 [8316d])
❯ yarn why @mikro-orm/nestjs
└─ services-core@workspace:.
   └─ @mikro-orm/nestjs@npm:5.1.5 [8316d] (via npm:^5.1.5 [8316d])
❯ yarn why @mikro-orm/core
└─ services-core@workspace:.
   └─ @mikro-orm/core@npm:5.6.3 [8316d] (via npm:^5.6.3 [8316d])
❯ node --version
v18.12.1
❯ yarn why typescript
├─ @ts-morph/common@npm:0.7.5
│  └─ typescript@patch:typescript@npm%3A4.1.6#~builtin<compat/typescript>::version=4.1.6&hash=bda367 (via patch:typescript@~4.1.3#~builtin<compat/typescript>)
│
├─ import-sort-parser-typescript@npm:6.0.0
│  └─ typescript@patch:typescript@npm%3A3.9.10#~builtin<compat/typescript>::version=3.9.10&hash=bda367 (via patch:typescript@^3.2.4#~builtin<compat/typescript>)
│
└─ services-core@workspace:.
   └─ typescript@patch:typescript@npm%3A4.7.3#~builtin<compat/typescript>::version=4.7.3&hash=bda367 (via patch:typescript@4.7.3#~builtin<compat/typescript>)
❯ yarn why mongodb
├─ @mikro-orm/mongodb@npm:5.6.3
│  └─ mongodb@npm:4.13.0 (via npm:4.13.0)
│
├─ @mikro-orm/mongodb@npm:5.6.3 [8316d]
│  └─ mongodb@npm:4.13.0 (via npm:4.13.0)
│
├─ mongodb-memory-server-core@npm:8.0.4
│  └─ mongodb@npm:4.2.0 (via npm:^4.1.3)
│
└─ services-core@workspace:.
   └─ mongodb@npm:4.2.2 (via npm:~4.2.0)
```

**Comment by B4nan** [comment, marked as answer]
Right, I know what's happening. The getter is supposed to be defined only for the "serialized primary key", which is a way the ORM handles [ObjectId/string duality in mongo](https://mikro-orm.io/docs/usage-with-mongo#objectid-and-string-id-duality) (so kinda the same as you are trying, but for ObjectIds). But it is used regardless of how you define the entity, because of how we setup the metadata defaults.

**Reply by xenoterracide** [reply]
I tried using `@SerializedPrimaryKey` too, it still replaces the getter. The reason this doesn't use it is I was hoping it would stop, this is *very* surprising behavior, and shouldn't be done, IMHO.

**Reply by xenoterracide** [reply]
can you explain why replacing this is an appropriate behavior and necessary?

**Reply by xenoterracide** [reply]
note: per the documentation you linked `console.log(author.id);  // returns '...id...'` no it doesn't, it returns `_id`

**Reply by B4nan** [reply]
> I tried using @SerializedPrimaryKey too, it still replaces the getter. 

Yes, that's exactly what the decorator is for - it defines the getter for you, you just define the property.

**Reply by xenoterracide** [reply]
then why if I don't use `@SerializedPrimaryKey`, does it still behave that way?

**Reply by B4nan** [reply]
Yeah that's a bug for sure, looking at that right now, but its apparently more complex than it seems.

**Reply by xenoterracide** [reply]
is it possible to move this discussion back to a bug then? I'm not sure if it is

**Reply by B4nan** [reply]
This should do it 

https://github.com/mikro-orm/mikro-orm/commit/700410075e0ff83207c17fe6a0413cd534208472

**Reply by xenoterracide** [reply]
thanks
